sample-clients/csharp/client.tests/ScoringClientTests.cs

```diff
@@ -90,6 +91,43 @@ public async Task Retries_transient_exceptions(StatusCode transientStatusCode)
             predictionServiceClientMock.Verify(x => x.PredictAsync(predictRequest), Times.Exactly(2));
         }
 
+        [Theory]
+        [InlineData(StatusCode.DeadlineExceeded, 2)]
+        [InlineData(StatusCode.Unavailable, 2)]
+        [InlineData(StatusCode.Aborted, 2)]
+        [InlineData(StatusCode.Internal, 2)]
+        public async Task Retries_transient_exceptions_until_retries_exhausted(StatusCode transientStatusCode, int retryCount)
+        {
+            var exception = new RpcException(new Status(transientStatusCode, string.Empty));
+
+            var expectedResultValues = new[] { 1f, 2f, 3f };
+
+            var outputTensorProto = new TensorProto { Dtype = DataType.DtFloat };
+            outputTensorProto.FloatVal.Add(expectedResultValues);
+
+            outputTensorProto.TensorShape = new TensorShapeProto();
+            outputTensorProto.TensorShape.Dim.Add(new TensorShapeProto.Types.Dim());
+            outputTensorProto.TensorShape.Dim[0].Size = 3;
+
+            var predictRequest = new PredictRequest();
+            var predictResponse = new PredictResponse();
+            predictResponse.Outputs.Add("output_alias", outputTensorProto);
+
+            var predictionServiceClientMock = new Mock<IPredictionServiceClient>();
```

Comment:
Maybe better to move creation of mocks to ctor and only leave mock configuration in here

Reply:
Refactoring client/Mocks into another method makes sense - imo outside the scope of this PR.